src/validation.cpp

```diff
@@ -1611,7 +1611,18 @@ static ThresholdConditionCache warningcache[VERSIONBITS_NUM_BITS];
 static unsigned int GetBlockScriptFlags(const CBlockIndex* pindex, const Consensus::Params& consensusparams) {
     AssertLockHeld(cs_main);
 
-    unsigned int flags = SCRIPT_VERIFY_NONE;
+
+    // BIP16 didn't become active until Apr 1 2012
+    // However, only one historical block violated the P2SH rules, so for simplicity, always leave P2SH
+    // on except for the one violating block.
+    bool fStrictPayToScriptHash = true;
+
+    if (pindex->phashBlock != nullptr && (*(pindex->phashBlock) == uint256S("0x00000000000002dc756eebf4f49723ed8d30cc28a5f108eb94b1ba88ac4f9c22") ||
+                (*(pindex->phashBlock) == uint256S("0x00000000dd30457c001f4095d208cc1296b0eed002427aa599874af7a432b105")))) { // testnet block!
+        fStrictPayToScriptHash = false;
+    }
+
+    unsigned int flags = fStrictPayToScriptHash ? SCRIPT_VERIFY_P2SH : SCRIPT_VERIFY_NONE;
 
     // Start enforcing P2SH (BIP16)
     if (pindex->nHeight >= consensusparams.BIP16Height) {
```

Comment:
fixup nit: in commit 3a76e3c962516cdf7d577aa3dcbe0251b94e5355: You probably meant to use ```if (fStrictPayToScriptHash) { flags |= SCRIPT_VERIFY_P2SH; }```, as BIP16Height was removed and thus fails to compile.